pkg/download/automation/automation.go

```diff
@@ -0,0 +1,111 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package automation
+
+import (
+	"encoding/json"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/maps"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/client/automation"
+	config "github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/coordinate"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter/value"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/template"
+	v2 "github.com/dynatrace/dynatrace-configuration-as-code/pkg/project/v2"
+)
+
+var automationTypesToResources = map[config.AutomationType]automation.ResourceType{
+	config.AutomationType{Resource: config.Workflow}:         automation.Workflows,
+	config.AutomationType{Resource: config.BusinessCalendar}: automation.BusinessCalendars,
+	config.AutomationType{Resource: config.SchedulingRule}:   automation.SchedulingRules,
+}
+
+// Downloader can be used to download automation resources/configs
+type Downloader struct {
+	client *automation.Client
+}
+
+// NewDownloader creates a new [Downloader] for automation resources/configs
+func NewDownloader(client *automation.Client) *Downloader {
+	return &Downloader{
+		client: client,
+	}
+}
+
+// Download downloads all automation resources for a given project
+// If automationTypes is given it will just download those types of automation resources
+func (d *Downloader) Download(projectName string, automationTypes ...config.AutomationType) (v2.ConfigsPerType, error) {
+
+	// if no specific automation types are given, take all
+	if len(automationTypes) == 0 {
+		automationTypes = maps.Keys(automationTypesToResources)
+	}
+
+	// download automation resources
+	configsPerType := make(v2.ConfigsPerType)
+	for _, at := range automationTypes {
+		resource, ok := automationTypesToResources[at]
+		if !ok {
+			log.Warn("No resource mapping for automation type %s found", at.Resource)
+			continue
+		}
+		// download all configs of resource
+		response, err := d.client.List(resource)
+		if err != nil {
+			return nil, err
+		}
+
+		// nothing to do if there were no configs
+		// continue with next resource
+		if len(response.Results) == 0 {
+			continue
+		}
+		log.Info("Downloaded %d objects for automation resource %s", len(response.Results), string(at.Resource))
+
+		// convert and collect all objects into a slice of configs.Config values to return
```

Comment:
nit: I'm not a big fan of these "general description comments" (specifically this one and line 59) - if the functionality is self-contained and needs a description, I'd prefer if you'd pull it into a function with a speaking name. (IMO might make sense here, for line 59 I'd personally drop the comment, downloading is exactly what I expect the automation.Download method to do)